### PR TITLE
docs: Navigation

### PR DESCRIPTION
**Pull request by Ombuweb**

### CODE REVIEW

File: content/guide/ui/navigation.md [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+---
+<!--TODO: Add Preview -->
+<!--TODO: Add flavors -->
+
+Navigation in NativeScript is enabled by the `Frame` class.
+
+## Using Frame
+
+### Navigating to another page
+To navigate to a page, first get the desired Frame instance. Then  for a simple navigation, call the instance's [navigate()](#navigate) method passing it the path of the page to navigate to.
```
Comment (SeanKelly369) [inline content/guide/ui/navigation.md]: How about ->

For page navigation, first obtain the desired Frame instance.
For a simple navigation, call the instance's navigate() method, with a string argument consisting of the desired path to navigate to.

File: content/guide/ui/navigation.md [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+frame.navigate("~/pages/details/details-page")
+```
+
+:::tip Note
+For a complete navigation example, have a look at [Setup navigation from home to details component](/tutorials/build-a-master-detail-app-with-plain-typescript#setup-navigation-from-home-to-details-component).
+:::
+
+### Getting a Frame instance
+
+The following are some of the ways to obtain an instance of the Frame class:
```
Comment (SeanKelly369) [inline content/guide/ui/navigation.md]: The following are some of the ways in which to obtain an instance of the Frame class:

File: content/guide/ui/navigation.md [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+- `always`
+- `never` (hides the action bar for all pages)
+
+---
+### navigate()
+This method has the following overloads:
+- `navigate(pageFileName: string)` - navigates to the page instance in the specified file.
+- `navigate(create: () => Page)` - creates a new Page instance using the provided callback and navigates to that instance.
+
+- `navigate(entry: NavigationEntry)` is for a more customized navigation. It allows you to pass data to another page or set transition animation, etc.
```
Comment (SeanKelly369) [inline content/guide/ui/navigation.md:189]: It provides the ability  to pass data to another page, along with optionally setting a transition animation.

etc?  What are the other options?

File: content/guide/ui/navigation.md [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+
+### goBack()
+
+```ts
+frame.goBack(to)
+```
+Navigates back using the navigation hierarchy (if any). Updates the Frame stack as needed. This method will start from the topmost Frame and will recursively search for an instance that has the [canGoBack](#cangoback) operation available.
```
Comment (SeanKelly369) [inline content/guide/ui/navigation.md]: Navigates back using the navigation hierarchy, contained within a Frame stack where the top value is popped.

**Review by SeanKelly369 — CHANGES_REQUESTED** [review]

File: content/guide/ui/navigation.md (hunk 1 above)
Comment (NathanWalker) [inline content/guide/ui/navigation.md]:
```suggestion
For page navigation, first obtain the desired Frame instance.
For a simple navigation, call the instance's navigate() method, with a string argument consisting of the desired path to navigate to.
```

File: content/guide/ui/navigation.md [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+frame.navigate("~/pages/details/details-page")
+```
+
+:::tip Note
+For a complete navigation example, have a look at [Setup navigation from home to details component](/tutorials/build-a-master-detail-app-with-plain-typescript#setup-navigation-from-home-to-details-component).
+:::
+
+### Getting a Frame instance
+
+The following are some of the ways to obtain an instance of the Frame class:
```
Comment (NathanWalker) [inline content/guide/ui/navigation.md]:
```suggestion
The following are some of the ways in which to obtain an instance of the Frame class:
```

File: content/guide/ui/navigation.md [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+- `navigate(create: () => Page)` - creates a new Page instance using the provided callback and navigates to that instance.
+
+- `navigate(entry: NavigationEntry)` is for a more customized navigation. It allows you to pass data to another page or set transition animation, etc.
+
+### goBack()
+
+```ts
+frame.goBack(to)
+```
+Navigates back using the navigation hierarchy (if any). Updates the Frame stack as needed. This method will start from the topmost Frame and will recursively search for an instance that has the [canGoBack](#cangoback) operation available.
```
Comment (NathanWalker) [inline content/guide/ui/navigation.md]:
```suggestion
Navigates back using the navigation hierarchy, contained within a Frame stack where the top value is popped.
```

**Review by SeanKelly369 — APPROVED** [review]

File: content/guide/ui/navigation.md (hunk 1 above)
Comment (NathanWalker) [inline content/guide/ui/navigation.md]:
```suggestion
For page navigation, first obtain the desired Frame instance.
For a simple navigation, call the instance's `navigate` method with a string argument consisting of the desired path to navigate to.
```

File: content/guide/ui/navigation.md (hunk 2 above)
Comment (NathanWalker) [inline content/guide/ui/navigation.md]:
```suggestion
The following are some of the ways in which to obtain an instance of the Frame class:
```

File: content/guide/ui/navigation.md (hunk 4 above)
Comment (NathanWalker) [inline content/guide/ui/navigation.md]:
```suggestion
Navigates back using the navigation stack contained within the topmost Frame and will recursively navigate back with each Frame instance which `canGoBack` returns true.
```

**Review by NathanWalker — APPROVED** [review]

**Pull request merged**